src/primo/browzine-primo-adapter.js

```diff
@@ -247,21 +247,23 @@ browzine.primo = (function() {
         if(browzineWebLink && browzineEnabled) {
           var template = buildTemplate(scope, browzineWebLink);
           element.append(template);
-          $scope.$apply();
         }
 
         if(coverImageUrl && browzineEnabled) {
-          setTimeout(function() {
-            var elementParent = getElementParent(element);
-            var coverImages = elementParent.querySelectorAll("prm-search-result-thumbnail-container img");
-
-            Array.prototype.forEach.call(coverImages, function(coverImage) {
-              coverImage.src = coverImageUrl;
-              $scope.$apply();
+          (function poll(){
+            setTimeout(function(){
+              var elementParent = getElementParent(element);
+              var coverImages = elementParent.querySelectorAll("prm-search-result-thumbnail-container img");
+
+              if(coverImages[0].className.indexOf("fan-img") > -1) {
+                Array.prototype.forEach.call(coverImages, function(coverImage) {
+                  coverImage.src = coverImageUrl;
+                });
+              } else {
+                poll();
```

Comment:
How frequently does this get called now?  Seems that we've got `setTimeout` being called without a time, and [MDN says](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout) it will do it "as soon as possible."

How much CPU does this take up?  I'm imagining this is using a decent amount of system resources...

Reply:
All good questions @karlbecker! The short answer is this will get called as many times as it needs to - No more, no less. The longer answer, of course, is the answer to this question is going to depend on a number of factors: Including the users network bandwidth, latency, their device's CPU, basically a host of things we can't control.

On my network and machine, I've seen Primo take anywhere from 1 to 3 (max) of these iterations to finally update their UI. Someone in Europe or New Zealand could have more iterations. Will they notice anything different, assuming their internet connection is generally just slow? Probably not. But whenever Primo updates their UI, we do the Journal Cover enhancements and the iterations stop.

We could add time to the intervals by setting a timeout value, of course, but the next customer support email we get about this will read something along the lines of the Journal Cover updates being "choppy", etc. :)

The key is what we have here is the fastest these Journal Cover enhancements can get. I'm comfortable trying out what we have here and then iterating on any feedback we get.

Reply:
Fascinating stuff!

![image](https://user-images.githubusercontent.com/376230/42663513-cad73e4c-8603-11e8-965e-49cac6c87b3a.png)

https://developer.mozilla.org/en-US/docs/Web/API/Frame_Timing_API
http://www.david-amador.com/2011/08/hurry-you-only-have-16-ms-to-render-everything/
https://www.quora.com/How-many-frames-per-second-can-a-browser-render-HTML

Reply:
<img width="726" alt="screen shot 2018-07-12 at 5 50 04 pm" src="https://user-images.githubusercontent.com/1531215/42663649-0ba8ab84-85fc-11e8-8f7e-f4990bbe818a.png">

Ooooooo

<img width="710" alt="screen shot 2018-07-12 at 5 50 31 pm" src="https://user-images.githubusercontent.com/1531215/42663665-176dd372-85fc-11e8-9287-5c600d8731ad.png">

Awwwwww. That could have been really cool. But 16 or 15ms should still cover all 99% of our users 😄

Reply:
Ok, we're pinned to 60fps now, but now ya got me wondering if we can use [requestAnimationFrame](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) here...?